### PR TITLE
Track menu, purge: allow to hide further success popups in the current session

### PR DESCRIPTION
**Pull request by ronso0**
adds a checkbox to the success popup
"Don't show again during this session"

Same method as for "Don't ask again" in the mapping overwrite dialog.

### CODE REVIEW

File: src/widget/wtrackmenu.h [hunk 1]
```diff
@@ -365,6 +365,8 @@ class WTrackMenu : public QMenu {
 
     Features m_eActiveFeatures;
     const Features m_eTrackModelFeatures;
+
+    static bool s_showPurgesuccessPopup;
```
Comment (cr7pt0gr4ph7) [inline src/widget/wtrackmenu.h:369]:
```suggestion
    static bool s_showPurgeSuccessPopup;
```

File: src/widget/wtrackmenu.cpp [hunk 2]
```diff
@@ -2422,47 +2424,56 @@ void WTrackMenu::slotRemoveFromDisk() {
         // Purge only those tracks whose files have actually been deleted.
         m_pLibrary->trackCollectionManager()->purgeTracks(tracksToPurge);
 
-        // Show purge summary message
-        QMessageBox msgBoxPurgeTracks;
-        msgBoxPurgeTracks.setIcon(QMessageBox::Information);
-        QString msgTitle;
-        QString msgText;
-        if (m_pTrackModel) {
+        if (s_showPurgesuccessPopup) {
```
Comment (cr7pt0gr4ph7) [inline src/widget/wtrackmenu.cpp:2427]:
```suggestion
        if (s_showPurgeSuccessPopup) {
```

File: src/widget/wtrackmenu.cpp [hunk 3]
```diff
@@ -94,6 +94,8 @@ void storeActionTextAndScaleInProperties(QAction* pAction, const double scale) {
 
 } // namespace
 
+bool WTrackMenu::s_showPurgesuccessPopup = true;
```
Comment (cr7pt0gr4ph7) [inline src/widget/wtrackmenu.cpp:97]:
```suggestion
bool WTrackMenu::s_showPurgeSuccessPopup = true;
```

File: src/widget/wtrackmenu.cpp [hunk 4]
```diff
@@ -2422,47 +2424,56 @@ void WTrackMenu::slotRemoveFromDisk() {
...
+            QString msgTitle;
+            QString msgText;
+            if (m_pTrackModel) {
 #if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
-            msgTitle = tr("Track Files Deleted");
+                msgTitle = tr("Track Files Deleted");
 #else
-            msgTitle = tr("Track Files Moved To Trash");
+                msgTitle = tr("Track Files Moved To Trash");
 #endif
```
Comment (cr7pt0gr4ph7) [inline src/widget/wtrackmenu.cpp:2438]: The `#if` for Qt5 can be removed, now that we have fully migrated to Qt6 (at least to my knowledge, correct me if I'm wrong)

```suggestion
                msgTitle = tr("Track Files Moved To Trash");
```

Also, the `#if` condition was already wrong/exactly backwards - compare with:
https://github.com/mixxxdj/mixxx/blob/edae0ca3b2fc36c51d9f310f341fb2e4bbea7d26/src/widget/wtrackmenu.cpp#L341-L345

File: src/widget/wtrackmenu.cpp [hunk 5]
```diff
@@ -2422,47 +2424,56 @@ void WTrackMenu::slotRemoveFromDisk() {
...
-                    tr("%1 track files were moved to trash and purged "
-                       "from the Mixxx database.")
+                        tr("%1 track files were moved to trash and purged "
+                           "from the Mixxx database.")
 #else
-                    tr("%1 track files were deleted from disk and purged "
-                       "from the Mixxx database.")
+                        tr("%1 track files were deleted from disk and purged "
+                           "from the Mixxx database.")
 #endif
```
Comment (cr7pt0gr4ph7) [inline src/widget/wtrackmenu.cpp:2446]: The `#if` for Qt5 can be removed here, too (though at least the `#if` condition is correct here):
```suggestion
                        tr("%1 track files were moved to trash and purged "
                           "from the Mixxx database.")
```
 
 https://github.com/mixxxdj/mixxx/blob/dbb9610fabf78b5957194ad89867f4896eff2743/src/widget/wtrackmenu.cpp#L2375C2-L2379C7

File: src/widget/wtrackmenu.cpp [hunk 6]
```diff
@@ -2422,47 +2424,56 @@ void WTrackMenu::slotRemoveFromDisk() {
...
 #else
-            msgTitle = tr("Track File Deleted");
-            msgText = tr(
-                    "Track file was deleted from disk and purged "
-                    "from the Mixxx database.");
+                msgTitle = tr("Track File Deleted");
+                msgText = tr(
+                        "Track file was deleted from disk and purged "
+                        "from the Mixxx database.");
 #endif
```
Comment (cr7pt0gr4ph7) [inline src/widget/wtrackmenu.cpp:2462]:
```suggestion
                msgTitle = tr("Track File Moved To Trash");
                msgText = tr(
                        "Track file was moved to trash and purged "
                        "from the Mixxx database.");
```

File: src/widget/wtrackmenu.cpp [hunk 7]
```diff
@@ -2422,47 +2424,56 @@ void WTrackMenu::slotRemoveFromDisk() {
...
+            notAgainCB.setCheckState(Qt::Unchecked);
+            msgBoxPurgeTracks.setWindowTitle(msgTitle);
+            msgBoxPurgeTracks.setText(msgText);
+            msgBoxPurgeTracks.setTextFormat(Qt::RichText);
+            msgBoxPurgeTracks.setCheckBox(&notAgainCB);
+            msgBoxPurgeTracks.setStandardButtons(QMessageBox::Ok);
+            msgBoxPurgeTracks.exec();
+
+            if (notAgainCB.isChecked()) {
+                s_showPurgesuccessPopup = false;
```
Comment (cr7pt0gr4ph7) [inline src/widget/wtrackmenu.cpp:2474]:
```suggestion
                s_showPurgeSuccessPopup = false;
```

**Review by cr7pt0gr4ph7 — CHANGES_REQUESTED** [review]
I'd suggest renaming the variable to `s_showPurgeSuccessPopup`, and also removing the Qt5 workarounds while already at it. Otherwise LGTM.

**Comment by ronso0** [comment]
Thanks. I fixed the static's name.

Also, I unified the Qt version switches.
Though, IMO this is not the place/time to remove Qt5 paths. Let's do this in one flush as soon as the core team agrees to drop Qt5 support entirely.

**Review by cr7pt0gr4ph7 — APPROVED** [review]
Looks good to me :+1:

**Review by daschuer — APPROVED** [review]
Thank you.

**Pull request merged**